src/components/shared/snackNotification.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <div id="ConnectionStatusSnack-ctn">
+    <v-snackbar v-model="visibility">
+      <span id="snack-message">
+        {{ message }}
+      </span>
+
+      <template #action="{ attrs }">
+        <div
+          v-bind="attrs"
+          id="close-icon"
+          @click="handleClose"
+        >
+          <Button
+            icon="close"
+            color="white"
+          />
+        </div>
+      </template>
+    </v-snackbar>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex'
+import AmbButton from '@/components/shared/Button'
+import { PEER_CONNECTED_NOTIFICATION, PEER_DISCONNECTED_NOTIFICATION, PEER_CONNECTING_NOTIFICATION } from '@/components/utils'
+
+export default {
+  name: 'ConnectionStatusSnack',
+  data: () => ({
+    visibility: true,
+    message: 'Connecting to Ambianic Edge device'
+  }),
+  components: {
+    Button: AmbButton
+  },
+  created () {
+    this.setConnectionStatusNotification()
+  },
+  computed: {
+    ...mapState({
+      peerConnectionStatus: state => state.pnp.peerConnectionStatus
+    })
+  },
+  methods: {
+    handleClose () {
+      this.visibility = false
+    },
+    setConnectionStatusNotification () {
+      switch (this.peerConnectionStatus) {
+        case 'PEER_CONNECTING':
+          this.visibility = this.message !== PEER_CONNECTING_NOTIFICATION
```

Comment:
@vickywane this repeated check should not be necessary. If the peer state machine fires the same state message multiple times, then something is wrong with the state machine.

Reply:
This condition was to handle scenarios when a user navigates to a new screen and the `setConnectionStatusNotification` function is (re)fired.